### PR TITLE
Add Siddhi Apps REST Api

### PR DESCRIPTION
**Pull request by LakshanSS**
>wso2.dashboard.datasearch:
>&nbsp;&nbsp;&nbsp;workerList:
>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- localhost:9443
>&nbsp;&nbsp;&nbsp;workerAccessCredentials:
>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;username: admin
>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;password: admin

>Send GET request to https://localhost:9643/portal/apis/datasearch/siddhi-apps to get siddhi apps

>Send GET request to https://localhost:9643/portal/apis/datasearch/siddhi-apps/siddhiAppName to get list of @store annotated elements

>To grant permission for user roles other than admin role, role id should be added in roleIdList as below

>wso2.dashboard.datasearch:
>&nbsp;&nbsp;&nbsp;workerList:
>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- localhost:9443
>&nbsp;&nbsp;&nbsp;workerAccessCredentials:
>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;username: admin
>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;password: admin
>&nbsp;&nbsp;&nbsp;roleIdList:
>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&nbsp;2

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/pom.xml [hunk 1]
```diff
@@ -0,0 +1,103 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Niveathika92) [inline components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/pom.xml]: License year to be changed to 2019

File: components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/src/main/java/org/wso2/carbon/siddhi/apps/api/rest/impl/SiddhiAppsApiServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    public Response getSiddhiApps(Request request) {
+        if (getUserName(request) != null && !permissionProvider.hasPermission(getUserName(request), new
+                Permission(PERMISSION_APP_NAME, VIEW_SIDDHI_APP_PERMISSION_STRING))) {
+            return Response.status(Response.Status.UNAUTHORIZED).entity("Insufficient permissions to view Siddhi Apps" +
+                    "for user " + getUserName(request))
+                    .build();
+        }
+
+        getSiddhiAppsFromWorkers();
+        List<String> siddhiAppList = new ArrayList<>(SiddhiAppsDataHolder.getInstance().getSiddhiAppMap().keySet());
```
Comment (Niveathika92) [inline components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/src/main/java/org/wso2/carbon/siddhi/apps/api/rest/impl/SiddhiAppsApiServiceImpl.java]: Use java streams fully,
```suggestion
        List<String> collect = SiddhiAppsDataHolder.getInstance().getSiddhiAppMap().keySet().stream().sorted().collect(Collectors.toList());
```

File: components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/src/main/java/org/wso2/carbon/siddhi/apps/api/rest/impl/SiddhiAppsApiServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            return Response.status(Response.Status.UNAUTHORIZED).entity("Insufficient permissions to view Siddhi " +
+                    "Apps for user " + getUserName(request)).build();
+        }
+
+        if (SiddhiAppsDataHolder.getInstance().getSiddhiAppMap().isEmpty()) {
+            getSiddhiAppsFromWorkers();
+        }
+
+        for (Map.Entry<String, List<SiddhiStoreElement>> entry :
+                SiddhiAppsDataHolder.getInstance().getSiddhiAppMap().entrySet()) {
```
Comment (Niveathika92) [inline components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/src/main/java/org/wso2/carbon/siddhi/apps/api/rest/impl/SiddhiAppsApiServiceImpl.java]:
```suggestion
                SiddhiAppsDataHolder.getInstance().getSiddhiAppMap().get(appName)) {
```
Use a simple get, we do not need to iterate through all siddhi app entries

**Comment by Niveathika92** [comment]
@LakshanSS Please use the PR template fully, the approach is how you have implemented, while here you have described the usage which should go under "Documentation".

File: components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/pom.xml (hunk 1 above)
Comment (LakshanSS) [reply components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/pom.xml]: Changed in 87c49f3

File: components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/src/main/java/org/wso2/carbon/siddhi/apps/api/rest/impl/SiddhiAppsApiServiceImpl.java (hunk 2 above)
Comment (LakshanSS) [reply components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/src/main/java/org/wso2/carbon/siddhi/apps/api/rest/impl/SiddhiAppsApiServiceImpl.java]: Fixed in 87c49f3

File: components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/src/main/java/org/wso2/carbon/siddhi/apps/api/rest/impl/SiddhiAppsApiServiceImpl.java (hunk 3 above)
Comment (LakshanSS) [reply components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/src/main/java/org/wso2/carbon/siddhi/apps/api/rest/impl/SiddhiAppsApiServiceImpl.java]: Fixed in 87c49f3

File: components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/src/main/java/org/wso2/carbon/siddhi/apps/api/rest/impl/SiddhiAppsApiServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    public Response getSiddhiApps(Request request) {
+        if (getUserName(request) != null && !permissionProvider.hasPermission(getUserName(request), new
+                Permission(PERMISSION_APP_NAME, VIEW_SIDDHI_APP_PERMISSION_STRING))) {
+            return Response.status(Response.Status.UNAUTHORIZED).entity("Insufficient permissions to view Siddhi Apps" +
+                    "for user " + getUserName(request))
+                    .build();
+        }
+
+        getSiddhiAppsFromWorkers();
+        List<String> siddhiAppList = new ArrayList<>(SiddhiAppsDataHolder.getInstance().getSiddhiAppMap().keySet()
```
Comment (Niveathika92) [inline components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/src/main/java/org/wso2/carbon/siddhi/apps/api/rest/impl/SiddhiAppsApiServiceImpl.java]: Stream collect will return an array list. Need not initate again

**Pull request merged**